### PR TITLE
Make tidy errors red

### PR DESCRIPTION
**Pull request by Noratrieb**
This makes it easier to see them (and makes people go owo).

I also changes the error codes check to not print too many things and use `tidy_error`.

r? @jyn514

### CODE REVIEW

**Review by jyn514 — COMMENTED** [review]
before:
```
tidy check
Checking which error codes lack tests...
* 633 error codes
* highest error code: E0791
tidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0790.md: empty file
tidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0790.md: leading newline
* 393 features
`/home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0790.md` doesn't use its own error code in compile_fail example
Found 1 error(s) in error codes
Done!
some tidy checks failed
```

after:
```
tidy check
* 633 error codes
* highest error code: E0791
tidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0790.md: empty file
tidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0790.md: leading newline
* 393 features
tidy error: `/home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0790.md` doesn't use its own error code in compile_fail example
some tidy checks failed
```

The changes in text look reasonable, but I'm not seeing them show up as red? Am I missing something?

image

**Review by Noratrieb — COMMENTED** [review]
whoops, accidentally buffered this comment instead of directly sending it

File: src/tools/tidy/src/deps.rs [hunk 1]
```diff
@@ -50,6 +50,7 @@ const EXCEPTIONS: &[(&str, &str)] = &[
     ("similar", "Apache-2.0"),       // cargo (dev dependency)
     ("normalize-line-endings", "Apache-2.0"), // cargo (dev dependency)
     ("dissimilar", "Apache-2.0"),    // rustdoc, rustc_lexer (few tests) via expect-test, (dev deps)
+    ("supports-color", "Apache-2.0"), // tidy
```
[the remark this answers is not in the document]
Comment (jyn514) [reply src/tools/tidy/src/deps.rs:53]: nah, this seems fine :)

File: src/tools/tidy/Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,7 @@ miropt-test-tools = { path = "../miropt-test-tools" }
 lazy_static = "1"
 walkdir = "2"
 ignore = "0.4.18"
+owo-colors = { version = "3.5.0", features = ["supports-colors"] }
```
Comment (Mark-Simulacrum) [inline src/tools/tidy/Cargo.toml:14]: How did we pick this library? Cargo (for example) is using termcolor; it seems like avoiding a different library for coloring might be better.

Comment (Nilstrieb) [reply src/tools/tidy/Cargo.toml:14]: i just picked a random one. but it probably makes sense to switch to termcolor, i'll do that

Comment (Nilstrieb) [reply src/tools/tidy/Cargo.toml:14]: How should we do the color support detection? Should I just use `supports-color` as now or reimplement some primitive detection myself?

Comment (jyn514) [reply src/tools/tidy/Cargo.toml:14]: `!CiEnv::is_ci()`, maybe?

Comment (Nilstrieb) [reply src/tools/tidy/Cargo.toml]: Actually, rust already depends on `owo-colors` through miri->ui_test->color-eyre. But I changed tidy to use termcolor anyways.

File: src/tools/tidy/src/lib.rs [hunk 3]
```diff
@@ -26,18 +30,27 @@ macro_rules! t {
...
+
+    *bad = true;
+
+    let mut stderr = StandardStream::stdout(ColorChoice::Auto);
+    stderr.set_color(ColorSpec::new().set_fg(Some(Color::Red)))?;
+
+    write!(&mut stderr, "tidy error")?;
+    stderr.set_color(&ColorSpec::new())?;
+
+    write!(&mut stderr, ": {args}")?;
```
Comment (jyn514) [inline src/tools/tidy/src/lib.rs:50]: needs to be `writeln`
```
tidy check
* 633 error codes
* highest error code: E0791
tidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0760.md: leading newlinetidy error: /home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0760.md: too many trailing newlines (2)* 393 features
some tidy checks failed
tidy error: `/home/gh-jyn514/rust2/compiler/rustc_error_codes/src/error_codes/E0760.md` doesn't use its own error code in compile_fail exampleBuild completed unsuccessfully in 0:00:10
```
```suggestion
    writeln!(&mut stderr, ": {args}")?;
```

**Comment by jyn514** [comment]
Looks great, thanks! r=me with CI passing :)

**Pull request merged**